services/MerchStoreService.ts

```diff
@@ -100,32 +102,49 @@ export default class MerchStoreService {
     const item = await this.transactions.readOnly(async (txn) => Repositories
       .merchStoreItem(txn)
       .findByUuid(uuid));
-    if (!item) throw new NotFoundError('Item not found');
+    if (!item) throw new NotFoundError(NotFoundErrors.MERCH_ITEM);
     return item.getPublicMerchItem();
   }
 
   public async createItem(item: MerchItem): Promise<MerchandiseItemModel> {
     return this.transactions.readWrite(async (txn) => {
+      if (this.hasMultipleOptionTypes(item.options)) throw new UserError(UserErrors.MULTIPLE_MERCH_OPTION_TYPES);
+      if (!item.hasVariants && item.options.length > 1) {
+        throw new UserError(UserErrors.NO_ITEM_VARIANTS_BUT_MULTIPLE_OPTIONS);
+      }
+
       const collection = await Repositories.merchStoreCollection(txn).findByUuid(item.collection);
-      if (!collection) throw new NotFoundError('Collection not found');
+      if (!collection) throw new NotFoundError(NotFoundErrors.MERCH_COLLECTION);
+
       const merchItemRepository = Repositories.merchStoreItem(txn);
       const merchItem = MerchandiseItemModel.create({ ...item, collection });
       await merchItemRepository.upsertMerchItem(merchItem);
       return merchItemRepository.findByUuid(merchItem.uuid);
     });
   }
 
+  private hasMultipleOptionTypes(options: MerchItemOption[] | MerchandiseItemOptionModel[] | MerchItemOptionEdit[]) {
+    for (let i = 1; i < options.length; i += 1) {
```

Comment:
This errors if an item only has a single variant. An alternative could be taking the set of all the metadata types and returning based on set size.

Reply:
Doesn't the loop not get executed if `options.length === 1`? Which means the function returns `false` and the errors wouldn't be thrown?

Reply:
Ah yeah, you're right.

Reply:
I personally prefer using sets for things like this since it's simpler but the loop works too.

Reply:
I think sets make more sense in this case, but it also might get weird since `metadata` could be null/undefined, so we shouldn't add anything to the set if that's the case, but then we'd need to check if the set is of size 0 or 1 (i.e. either single variant or multi variant with 1 option type), which makes it pretty much as complicated as the loop but without constant space.

Reply:
So it might just be better to stick with the loop even though it's a little more archaic

Reply:
This function is only called for items with variants enabled, so we should be catching undefined `metadata` before calling this.

Reply:
I don't think that's always the case - the function is ran if the check above it passes:
```ts
if (!item.hasVariantsEnabled && item.options.length > 1) {
  throw new UserError(UserErrors.VARIANTS_DISABLED_MULTIPLE_OPTIONS);
}
if (this.hasMultipleOptionTypes(item.options)) throw new UserError(UserErrors.MULTIPLE_MERCH_OPTION_TYPES);
```
so `item` could have `hasVariantsEnabled = false` and only 1 option (thus having potentially empty metadata), or it could have `hasVariantsEnabled = true` and any number of options.

Reply:
We should only be calling `hasMultipleOptionTypes` for items with variants enabled, so
```typescript
if (item.hasVariantsEnabled && this.hasMultipleOptionTypes(item.options)) { ... }
```
should work, since if variants is enabled for an item its options should always have metadata.